document-viewer/src/main/java/org/ebookdroid/common/provider/DocumentContentProvider.java

```diff
@@ -0,0 +1,54 @@
+package org.ebookdroid.common.provider;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+public class DocumentContentProvider extends ContentProvider {
+
+    @Override
+    public ParcelFileDescriptor openFile(Uri uri, String mode) throws FileNotFoundException {
+        File file = new File(uri.getPath());
+
+        if (file.exists()) {
+            return ParcelFileDescriptor.open(file, ParcelFileDescriptor.MODE_READ_ONLY);
+        }
+
+        throw new FileNotFoundException(uri.getPath());
+    }
+
+    @Override
+    public boolean onCreate() {
+        return true;
+    }
+
+    @Override
+    public String getType(Uri uri) {
+        return uri.getQuery();
+    }
```

Comment:
According to http://developer.android.com/guide/topics/providers/content-provider-creating.html#FileMIMETypes the ContentProvider should also implement `getStreamTypes()`, if I am understanding that correctly